(feat): Adr search collator module (#549)
backstage/community-plugins: workspaces/adr/.changeset/healthy-horses-wait.md

---

> '@backstage-community/plugin-adr-backend': patch

Reviewer: Since this still a pre-major version package, feel free to make this a breaking change (minor bump) and go ahead and just remove these types from this package - just make sure to indicate it here with a `BREAKING CHANGE` note

Reviewer: and also don't forget to remove related deps from this package that are no longer necessary

Reviewer: right currently it is a patch but I think its ok to actually make this a breaking minor bump since this plugin is pre major v1 so there's no obligation to provide a deprecation period

Author: Let me make that change then. I'll just remove the dependencies from the backend package in favor of the module instead, and add a tiny breaking change message in the release notes

---

> Deprecated the `search` types, which should be now imported from "@backstage-community/search-backend-module-adr" instead

Reviewer 2: ```suggestion
The `DefaultAdrCollatorFactory` has been deprecated in favour of using the new `@backstage-community/search-backend-module-adr` module.
```

Author: Reworded the changelog. Also added a second paragraph with the types that have been moved

---

> '@backstage-community/plugin-adr-backend': minor

Reviewer 2: as I said in the other comment we should try to avoid the breaking change, making this as a patch

Author: Hi @vinzscam . I've made the changes again a patch for the backend module and added the deprecation for all the types now. I think this should be ready to go now :)
